internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/storagenode/ReadComputeValidationTest.java

```diff
@@ -657,4 +628,458 @@ public void testCountGroupByValueAggregation() throws Exception {
+
+      // Cast to AvroGenericReadComputeStoreClient to access computeAggregation method
+      AvroGenericReadComputeStoreClient<Integer, Object> computeStoreClient =
+          (AvroGenericReadComputeStoreClient<Integer, Object>) storeClient;
+
+      // Write test data to the store
+      pushRecordsToStore(topic, valuesByKey, veniceWriter, valueSerializer, 1);
+
+      Set<Integer> keySet = new HashSet<>(Arrays.asList(1));
+
+      TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, false, true, () -> {
+        // Test 1: Try to use a predicate on a non-existent field
+        Map<String, Predicate<String>> invalidFieldBuckets = new HashMap<>();
+        invalidFieldBuckets.put("invalid", Predicate.equalTo("test"));
+
+        try {
+          computeStoreClient.computeAggregation()
+              .countGroupByBucket(invalidFieldBuckets, "nonExistentField")
+              .execute(keySet)
+              .get();
+          Assert.fail("Should have thrown an exception for non-existent field");
+        } catch (Exception e) {
+          // Expected to throw an exception
+          System.out.println("Expected exception for non-existent field: " + e.getMessage());
+        }
+
+        // Test 2: Try to use a predicate type that doesn't match the field type
+        Map<String, Predicate<Integer>> typeMismatchBuckets = new HashMap<>();
+        typeMismatchBuckets.put("mismatch", IntPredicate.equalTo(25));
+
+        try {
+          computeStoreClient.computeAggregation()
+              .countGroupByBucket(typeMismatchBuckets, "salary") // salary is float, not int
+              .execute(keySet)
+              .get();
+          Assert.fail("Should have thrown an exception for type mismatch");
+        } catch (Exception e) {
+          // Expected to throw an exception
+          System.out.println("Expected exception for type mismatch: " + e.getMessage());
+        }
+
+        // Test 3: Try to use an unsupported predicate combination
+        Map<String, Predicate<Integer>> unsupportedBuckets = new HashMap<>();
+        // Create a deeply nested predicate that might not be supported
+        Predicate<Integer> complexPredicate = Predicate.and(
+            IntPredicate.greaterThan(20),
+            Predicate
+                .or(IntPredicate.equalTo(25), Predicate.and(IntPredicate.lowerThan(30), IntPredicate.greaterThan(15))));
+        unsupportedBuckets.put("complex", complexPredicate);
+
+        try {
+          computeStoreClient.computeAggregation().countGroupByBucket(unsupportedBuckets, "age").execute(keySet).get();
+          // This might succeed or fail depending on implementation
+          // We're testing that the system handles complex predicates gracefully
+          System.out.println("Complex predicate test completed");
+        } catch (Exception e) {
+          // If it fails, that's also acceptable
+          System.out.println("Complex predicate failed as expected: " + e.getMessage());
```

Comment:
Let's avoid using System.out.println to print out logs, is it possible we could verify the expection type or exception message?